### PR TITLE
Updated registrar sync command

### PR DESCRIPTION
**Pull request by alangsto**
## [MST-434](https://openedx.atlassian.net/browse/MST-434)

@edx/masters-devs-cosmonauts 

The first version of the command did not account for multiple roles within an existing organization group, so logic was included to update only the name of the role, and not the string corresponding to the role name. 

Logic is also included to account for existing organizations that may not have a `ReadOrganizationReports` role associated with it, because in the previous version, organizations that fell under that case would not have this new group/role created.

In order to first replicate this error, I made modifications to the testing file that is included in this PR. Once the additional logic was added, all tests passed with these new organization groups added.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -12,8 +12,10 @@
 # serve to show the default.
 
 import os
+
```
Comment (schenedx) [inline docs/conf.py]: #NIT can we not introduce these unnecessary spacing changes?

File: registrar/apps/core/management/commands/test/test_sync_with_discovery.py [hunk 2]
```diff
@@ -47,10 +48,15 @@ def setUpTestData(cls):
...
             role=OrganizationReadReportRole.name
         )
-        cls.other_org_read_reports_group = OrganizationGroupFactory(
-            name='{}_ReadOrganizationReports'.format(cls.other_org.key),
+        cls.org_read_enrollments = OrganizationGroupFactory(
+            name='{}_ReadEnrollments'.format(cls.org.key),
+            organization=cls.org,
+            role=OrganizationReadEnrollmentsRole.name
+        )
+        cls.other_org_read_enrollments = OrganizationGroupFactory(
```
Comment (schenedx) [inline registrar/apps/core/management/commands/test/test_sync_with_discovery.py:56]: This is not referenced anywhere.  Why do we need the `other_org_read_enrollments group`?

**Review by schenedx — COMMENTED** [review]
Can you explain why in stage, we had duplicated readReports groups created? And how your PR here fixes that issue on stage?

File: registrar/apps/core/management/commands/test/test_sync_with_discovery.py (hunk 2 above)
Comment (alangsto) [reply registrar/apps/core/management/commands/test/test_sync_with_discovery.py:56]: In order to replicate what happened on stage, I wanted to test three kinds of orgs: one with two groups (including a read reports group), one with no groups, and one that had existing groups that did not include a read reports group. `other_org_read_enrollments` allows me to test the last case and ensures that existing orgs with no read reports group will have one created.

**Comment by alangsto** [comment]
@schenedx there were duplicate readReports groups due to this function in the original:
```
def update_org_groups(self, updated_orgs):
        """
        Update the existing OrganizationGroups to match the up to date name of the organization
        """
        existing_org_groups = OrganizationGroup.objects.select_related('organization').filter(
            organization__discovery_uuid__in=[org.discovery_uuid for org in updated_orgs]
        )
        org_group_to_update = []
        for org_group in existing_org_groups:
            org_group.name = '{}_ReadOrganizationReports'.format(org_group.organization.key)
            org_group_to_update.append(org_group)

```

The line: `org_group.name = '{}_ReadOrganizationReports'.format(org_group.organization.key)` rewrites every org group name for the org groups that are in the query set, so a Mattx_ReadEnrollments would become Mattx_ReadOrganizationReports.

These two lines (https://github.com/edx/registrar/pull/366/files#diff-7253cbab64da1f3665d25be5d804b31cR169-R170) address this issue.

**Review by schenedx — APPROVED** [review]
👍

**Pull request merged**